tokio/src/executor/task/mod.rs

```diff
@@ -63,34 +65,43 @@ where
     T: Future + Send + 'static,
     S: Schedule,
 {
-    let raw = RawTask::new_background(task);
-    Task { raw }
+    Task {
+        raw: RawTask::new_background::<_, S>(task),
+        _p: PhantomData,
+    }
 }
 
 /// Create a new task with an associated join handle
-pub(crate) fn joinable<T, S>(task: T) -> (Task<S>, JoinHandle<T::Output, S>)
+pub(crate) fn joinable<T, S>(task: T) -> (Task<S>, JoinHandle<T::Output>)
```

Comment:
Can this really take an arbitrary `S`? Or is it expected to be the same `S` as some other `S`? If the latter, I'd like that specified in the doc for the function, and possibly even for the `fn` to be marked as `unsafe`.

Reply:
Reading the code more carefully, I think it really can be an arbitrary `S`. Nice.